homeassistant/components/switchbot_cloud/config_flow.py

```diff
@@ -0,0 +1,66 @@
+"""Config flow for SwitchBot via API integration."""
+
+from logging import getLogger
+from typing import Any
+
+from switchbot_api import CannotConnect, InvalidAuth, SwitchBotAPI
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY, CONF_API_TOKEN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN, ENTRY_TITLE
+
+_LOGGER = getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_TOKEN): str,
+        vol.Required(CONF_API_KEY): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
```

Comment:
I would suggest to actually just remove this function. You're not mutating data, you're only doing 1 call, that can be moved into the place you call the function. Might be a lot cleaner